modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -171,6 +170,17 @@ class SliceLayerImpl CV_FINAL : public SliceLayer
         else
             CV_Assert(outputs.size() == sliceRanges.size());
 
+        for (int i = 0; i < sliceRanges.size(); ++i)
+        {
+            int rangeSize = sliceRanges[i].size();
+            int dims = inpShape.dims();
+            if (rangeSize < dims) {
+                for (int j = rangeSize; j < dims; j++) {
+                    sliceRanges[i].push_back(Range(0, -1));
```

Comment:
Could you please describe what it does?

Reply:
sliceRanges[i].size() = axis + begin.size().